.github/workflows/pr-staticcheck.yaml

```diff
@@ -0,0 +1,71 @@
+name: Pull request review
+on:
+  pull_request
+jobs:
+  run-danger:
+    runs-on: ubuntu-latest
+    environment: dev
+    concurrency:
+      group: ${{ github.workflow }}-pr-staticcheck-danger-${{ github.head_ref || github.run_id }}
+      cancel-in-progress: true
+    steps:
+      - id: checkout
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab
+      - id: setup-node
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c
+        with:
+          node-version-file: '.node-version'
+      - run: '[ -z "$${{ secrets.DANGER_GITHUB_API_TOKEN }}" ] || yarn danger ci'
+  compile:
+    runs-on: ubuntu-latest
+    concurrency:
+      group: ${{ github.workflow }}-pr-staticcheck-compile-${{ github.head_ref || github.run_id }}
+      cancel-in-progress: true
+    steps:
+      - id: checkout
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab
+      - id: setup
+        uses: ./.github/actions/setup-composite
+      - id: run-tsc
+        run: yarn tsc:noemit
+  lint:
```

Comment:
Why do we separate each check (`lint`, `compile`, `prettify`, etc.) as separate jobs instead to put them as steps in a single job? Is it for the sake of concurrency? We want to run them in parallel? Anyway we have duplicated steps like checkout and setup-composite... Do we have a performance comparison between the two approaches (multiple jobs vs single job with multiple steps).

Reply:
If this scenario performs better (multiple jobs) why do we have the checkout step by its own instead of putting it inside our composite action?